Source/Common/ArgumentLoader.cpp

```diff
@@ -5,47 +5,260 @@
 #include <FEXCore/Config/ConfigOptions.inl>
+    [[likely]] default:
+      Loader->SetArg(Details->Option, SecondArg);
+      break;
+    }
+  }
 
-  Parser.add_option_group(CPUGroup);
-  Parser.add_option_group(EmulationGroup);
-  Parser.add_option_group(DebugGroup);
-  Parser.add_option_group(HacksGroup);
-  Parser.add_option_group(MiscGroup);
-  Parser.add_option_group(LoggingGroup);
+  void ParseArgument_BoolTrue(std::string_view Arg, std::string_view SecondArg, const OptionDetails* Details) {
+    Loader->SetArg(Details->Option, "1");
+  }
 
-  optparse::Values Options = Parser.parse_args(argc, argv);
+  void ParseArgument_BoolFalse(std::string_view Arg, std::string_view SecondArg, const OptionDetails* Details) {
+    Loader->SetArg(Details->Option, "0");
+  }
 
-  using int32 = int32_t;
-  using uint32 = uint32_t;
-#define AFTER_PARSE
+  void ParseArgument_StringEnum(std::string_view Arg, std::string_view SecondArg, const OptionDetails* Details) {
+    switch (Details->Option) {
+#define STR_ENUM_PARSE
 #include <FEXCore/Config/ConfigOptions.inl>
-  RemainingArgs = Parser.args();
-  ProgramArguments = Parser.parsed_args();
+    [[unlikely]] default:
+      ExitWithError(fextl::fmt::format("Unknown strenum argument: {}", Arg));
+      break;
+    }
+  }
+
+  constexpr static std::array<ParseArgHandler, FEXCore::ToUnderlying(ValueType::Max)> Parser = {
+    &ArgParser::ParseArgument_String,
+    &ArgParser::ParseArgument_BoolTrue,
+    &ArgParser::ParseArgument_BoolFalse,
+    &ArgParser::ParseArgument_StringEnum,
+    // Behaves the same as string but appends multiple.
+    &ArgParser::ParseArgument_String,
+  };
+
+  void ParseArgument(std::string_view Arg, std::string_view SecondArg, const OptionDetails* Details) {
+    (this->*Parser[FEXCore::ToUnderlying(Details->ParseType)])(Arg, SecondArg, Details);
+  }
+};
+
+static bool NeedsArg(const OptionDetails* Details) {
+  return Details->ParseType == ValueType::String || Details->ParseType == ValueType::StringEnum || Details->ParseType == ValueType::StringArray;
+}
+
+std::pair<fextl::vector<fextl::string>, fextl::vector<const char*>> ArgParser::Parse(int argc, char** argv) {
+  fextl::vector<fextl::string> RemainingArgs {};
+  fextl::vector<const char*> ProgramArguments {};
+
+  // Skip argv[0]
+  int ArgParsed = 1;
+  for (; ArgParsed < argc; ++ArgParsed) {
+    std::string_view Arg = argv[ArgParsed];
+
+    // Special case version and help
+    if (Arg == "--version") [[unlikely]] {
+      fextl::fmt::print("{}\n", _Version);
+      std::exit(0);
+    }
+
+    if (Arg == "-h" || Arg == "--help") [[unlikely]] {
+      PrintHelp();
+      std::exit(0);
+    }
+
+    if (Arg == "--") {
+      // Special case break. Remaining arguments get passed to guest.
+      ++ArgParsed;
+      break;
+    }
+
+    const bool IsShort = Arg.find("--", 0, 2) == Arg.npos && Arg.find("-", 0, 1) == 0;
+    const bool IsLong = Arg.find("--", 0, 2) == 0;
+
+    std::string_view ArgFirst {};
```

Comment:
This variable doesn't need to be declared before proper initialization, since the two branches don't converge in a path that still uses the variable.